internal/api/api.go

```diff
@@ -0,0 +1,124 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"log/slog"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/kyma-project/btp-manager/internal/api/vm"
+	clusterobject "github.com/kyma-project/btp-manager/internal/cluster-object"
+	servicemanager "github.com/kyma-project/btp-manager/internal/service-manager"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+type API struct {
+	cluster                 *client.Client
+	namespaceProvider       *clusterobject.NamespaceProvider
+	serviceInstanceProvider *clusterobject.ServiceInstanceProvider
```

Comment:
Reading Service Instances from the cluster is out of scope, `ServiceInstanceProvider` is a dependency to SM client  for retrieving secrets names referenced in SI resources. SM client will provide Service Instances information from Service Manager.